src/test/integration/README

```diff
@@ -0,0 +1 @@
+This directory contains end-to-end tests.
```

Comment:
For the context, I added it here because it reproduced an real-life issue I got while testing, that no other tests caught. So it's included mostly because I fixed this issue and wanted a test for this. But it's easy to move it to another PR.